test/helpers/name.go

```diff
@@ -34,3 +44,46 @@ func ObjectPrefixForTest(t *testing.T) string {
 func ObjectNameForTest(t *testing.T) string {
 	return AppendRandomString(ObjectPrefixForTest(t))
 }
+
+// AppendRandomString will generate a random string that begins with prefix.
+// This is useful if you want to make sure that your tests can run at the same
+// time against the same environment without conflicting.
+// This method will use "-" as the separator between the prefix and
+// the random suffix.
+// This method will seed rand with the current time when the package is initialized.
+func AppendRandomString(prefix string) string {
+	suffix := make([]byte, randSuffixLen)
+
+	for i := range suffix {
+		suffix[i] = letterBytes[rand.Intn(len(letterBytes))]
+	}
+
+	return strings.Join([]string{prefix, string(suffix)}, string(sep))
+}
+
+// MakeK8sNamePrefix converts each chunk of non-alphanumeric character into a single dash
+// and also convert camelcase tokens into dash-delimited lowercase tokens.
+func MakeK8sNamePrefix(s string) string {
```

Comment:
I've seen this methods before, and wanted to make sure they're not another set of duplicates.

Reply:
I'm pretty sure it's not, since initially it was I that moved this method to this package from `serving` :)